src/lazy/expanded/macro_evaluator.rs

```diff
@@ -1272,6 +1259,81 @@ mod tests {
         )
     }
 
+    #[test]
+    fn macros_close_over_dependencies() -> IonResult<()> {
+        eval_enc_expr(
+            r#"
+                // Define macro `double`
+                $ion_encoding::(
+                    (macro_table
+                        $ion
+                        (macro double ($x) ($ion::values $x $x))
+                    )
+                )
+
+                // `double` exists until the *end* of the encoding directive below. Define a new
+                // macro that depends on `double`.
+                $ion_encoding::(
+                    (macro_table
+                        (macro quadruple ($y)
+                            ($ion::values
+                                // Because `double` is in the active encoding module, we can refer
+                                // to it without qualification.
+                                (double $y)
+                                // We could also refer to it with a qualification.
+                                ($ion_encoding::double $y)))
+                    )
+                )
+
+                // At this point, only `quadruple` is accessible/addressable, but it still works.
+                (:quadruple "foo")
+            "#,
+            r#"
+                "foo"
+                "foo"
+                "foo"
+                "foo"
+            "#,
+        )
+    }
+
+    #[test]
+    fn reexporting_ion_encoding_makes_macros_local() -> IonResult<()> {
+        eval_enc_expr(
+            r#"
+                // Define macro `double`
+                $ion_encoding::(
+                    (macro_table
+                        $ion_encoding
+                        (macro double ($x) (values $x $x))
+                    )
+                )
+
+                $ion_encoding::(
+                    (macro_table
+                        $ion_encoding // Re-export the active encoding module's macros
```

Comment:
In this particular case, you could also write `(export double)`, right?

Reply:
Yes 👍